Sources/PriorityQueueModule/PriorityQueue.swift

```diff
@@ -0,0 +1,168 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+/// A double-ended priority queue built on top of `Heap`.
+public struct PriorityQueue<Value, Priority: Comparable> {
+  public typealias Pair = (value: Value, priority: Priority)
+
+  @usableFromInline
+  struct _Element: Comparable {
+    @usableFromInline let value: Value
+    let priority: Priority
+    let insertionCounter: UInt64
```

Comment:
Anyone have any thoughts on this approach? Is using `UInt64` overkill for this? Should we overflow back to 0 on insertion instead of trapping?